### PR TITLE
Broken link fixes

### PR DESCRIPTION
**Pull request by jlstevens**
This PR aims to fix the broken links identified in #1650.

I don't intend to worry about the broken links on ``build.holoviews.org`` as many of those are actually fine on ``holoviews.org``. At least to start with, I will focus on links I think are genuinely broken...

### CODE REVIEW

**Comment by jlstevens** [comment]
@philippjfr What link do you want to use for the matplotlib area_chart notebook? It currently points to Bokeh.

**Comment by jlstevens** [comment]
@philippjfr Please merge this PR first in case there are merge conflicts with #2507

I hope this PR will be finished soon...

**Comment by jlstevens** [comment]
@philippjfr Ready to review/merge.

File: doc/showcase/index.rst [hunk 1]
```diff
@@ -61,9 +61,9 @@ Third party libraries, simulators and toolkits that make use of
 HoloViews for easier visualization and analysis:
 
 * `ImaGen library <http://ioam.github.io/imagen>`_: Generate
-  HoloViews `Image <https://holoviews/reference/elements/bokeh/Image.html>`_
+  HoloViews `Image <https://holoviews.org/reference/elements/bokeh/Image.html>`_
   and 
-  `RGB <https://holoviews/reference/elements/bokeh/RGB.html>`_ 
+  `RGB <https://holoviews.org/reference/elements/bokeh/RGB.html>`_
```
Comment (philippjfr) [inline doc/showcase/index.rst]: How did this even happen?

**Comment by philippjfr** [comment]
Looks good.

**Pull request merged**